internal/driver/fetch_test.go

```diff
@@ -206,6 +206,138 @@ func TestFetch(t *testing.T) {
 	}
 }
 
+func TestFetchWithBase(t *testing.T) {
+	const path = "testdata/"
+	// set up testcases
+	type testcase struct {
+		sources         []string
+		bases           []string
+		normalize       bool
+		testDescription string
+		expectSamples   bool
+	}
+
+	testcases := []testcase{
+		{
+			[]string{
+				path + "cppbench.cpu",
+			},
+			[]string{
+				path + "cppbench.cpu",
+			},
+			false,
+			"not normalized, base is same as non-base",
+			false,
+		},
+		{
+			[]string{
```

Comment:
Nit: you could shift prepending the path prefix from the test cases to the code of the test. And make the array declarations here one liners then, it would make the source more compact, like

```
		{
			[]string{"cppbench.cpu"},
			[]string{"cppbench.cpu", "cppbench.cpu"},
			false,
			"description",
			true,
		}
```